core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -83,6 +89,11 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   def main(args: Array[String]) {
 
     info("Starting mirror maker")
+    println("WARNING: The default partition assignment strategy of mirror maker will change from 'range' to 'roundrobin' " +
```

Comment:
@vahidhashemian Ah yeah, what I mean is that you can actually specify both assignors in the consumer's configuration (partition.assignment.strategy accepts a list). The names of all assignment strategies will get forwarded to the coordinator, which will attempt to choose one which is shared by all members in the group. The idea was to use this in a rolling upgrade. For example, say you have consumers A and B which are using range assignment. You can change the configuration of consumer A to be "round-robin,range" and do a restart. The coordinator will continue to choose "range" since it's the only strategy both consumers. You can then change the configuration of consumer B to also be "round-robin,range" and then the coordinator will choose "round-robin" (preference is implied by the order in the list). Ideally, you would then do another round of rolling restarts in which you remove "range" from the list.

Sounds great, right? The problem is that supporting multiple assignment strategies carries some overhead in the rebalance protocol. We end up sending the topic subscription list once for each supported strategy. In some MM use cases (LI in particular), this can be a problem since the consumer might be subscribed to thousands of topics. Still, it might be a reasonable tradeoff since most users won't have so many topics and it provides an easier upgrade path.